awswrangler/opensearch/_utils.py

```diff
@@ -1,23 +1,33 @@
 """Amazon OpenSearch Utils Module (PRIVATE)."""
 
+import json
 import logging
 import re
-from typing import Any, Optional
+import time
+from typing import Any, Dict, List, Optional, Union
 
 import boto3
+import botocore
 from opensearchpy import OpenSearch, RequestsHttpConnection
 from requests_aws4auth import AWS4Auth
 
 from awswrangler import _utils, exceptions
 
 _logger: logging.Logger = logging.getLogger(__name__)
 
+_CREATE_COLLECTION_FINAL_STATUSES: List[str] = ["ACTIVE", "FAILED"]
+_CREATE_COLLECTION_WAIT_POLLING_DELAY: float = 1.0  # SECONDS
+
 
 def _get_distribution(client: OpenSearch) -> Any:
+    if getattr(client, "_serverless", False):
```

Comment:
Might be mistaken here but isn't the OpenSearch client the same for both serverless and none serverless? In which case the exisiting client.info... line should already do the job?

Reply:
This is a workaround to avoid calling `.info()` because info endpoint isn't available in OpenSearch serverless [(more here)](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-genref.html). Sadly there isn't an endpoint to query cluster status or to get the version even - only a few collection/index operations are permitted

Reply:
Let us ask the OpenSearch team as it feels like an ugly workaround

Reply:
Sure. 

An alternative at this point is to let it send a request and come back with `NotFoundError` but I think it's better to save on extra requests if we know it's going to fail.